Mathlib/Algebra/Ring/CentroidHom.lean

```diff
@@ -467,21 +467,44 @@ lemma centerToCentroid_apply (z : { x // x ∈ NonUnitalSubsemiring.center α })
     (centerToCentroid z) a = z * a := rfl
 
 lemma center_iff_op_centroid (a : α) :
-    a ∈ NonUnitalSubsemiring.center α ↔ L a = R a ∧ (L a) ∈ Set.range CentroidHom.toEnd := by
+    a ∈ NonUnitalSubsemiring.center α ↔ R a = L a ∧ (L a) ∈ RingHom.rangeS (toEndRingHom α) := by
   constructor
-  · exact fun ha ↦ ⟨AddMonoidHom.ext <| IsMulCentral.comm ha, ⟨centerToCentroid ⟨a, ha⟩, rfl⟩⟩
+  · exact fun ha ↦ ⟨AddMonoidHom.ext <| fun _ => (IsMulCentral.comm ha _).symm,
+      ⟨centerToCentroid ⟨a, ha⟩, rfl⟩⟩
   · rintro ⟨hc, ⟨T, hT⟩⟩
     have e1 (d : α) : T d = a * d := congr($hT d)
-    have e2 (d : α) : T d = d * a := congr($(hT.trans hc) d)
+    have e2 (d : α) : T d = d * a := congr($(hT.trans hc.symm) d)
     constructor
-    case comm => exact (congr($hc ·))
+    case comm => exact (congr($hc.symm ·))
     case left_assoc => simpa [e1] using (map_mul_right T · ·)
     case mid_assoc => exact fun b c ↦ by simpa [e1 c, e2 b] using
       (map_mul_right T b c).symm.trans <| map_mul_left T b c
     case right_assoc => simpa [e2] using (map_mul_left T · ·)
 
 end NonUnitalNonAssocSemiring
 
+section NonUnitalNonAssocCommSemiring
+
+variable [NonUnitalNonAssocCommSemiring α]
+
+local notation "L" => AddMonoid.End.mulLeft
+local notation "R" => AddMonoid.End.mulRight
+
+lemma center_iff_op_mul_commute (a : α) :
+    a ∈ NonUnitalSubsemiring.center α ↔ ∀ b : α, Commute (L b) (L a) := by
+  rw [center_iff_op_centroid, centroid_eq_centralizer_mulLeftRight, Subsemiring.mem_centralizer_iff,
+    AddMonoid.End.comm_mulRight_eq_mulLeft, Set.union_self]
+  simp_rw [true_and]
+  constructor
+  · intro h b
+    apply h (L b)
+    simp only [Set.mem_range, exists_apply_eq_apply]
+  · intro h _ ⟨b, hb⟩
+    rw [← hb]
+    exact h b
```

Comment:
```suggestion
  simp_rw [true_and]
  exact ⟨fun h b ↦ h (L b) ⟨b, rfl⟩, by rintro h _ ⟨b, rfl⟩; exact h b⟩
```
or, if you prefer: `aesop` works in places of all of this.

Reply:
Thanks, I've replaced this with `aesop`.